java/feature-layer-shapefile/src/main/java/com/esri/arcgisruntime/sample/featurelayershapefile/MainActivity.java

```diff
@@ -46,23 +48,23 @@ protected void onCreate(Bundle savedInstanceState) {
     mMapView.setMap(map);
 
     // load the shapefile with a local path
-    ShapefileFeatureTable shapefileFeatureTable = new ShapefileFeatureTable(
+    mShapefileFeatureTable = new ShapefileFeatureTable(
         getExternalFilesDir(null) + getString(R.string.shapefile_path));
 
-    shapefileFeatureTable.loadAsync();
-    shapefileFeatureTable.addDoneLoadingListener(() -> {
-      if (shapefileFeatureTable.getLoadStatus() == LoadStatus.LOADED) {
+    mShapefileFeatureTable.loadAsync();
+    mShapefileFeatureTable.addDoneLoadingListener(() -> {
+      if (mShapefileFeatureTable.getLoadStatus() == LoadStatus.LOADED) {
 
         // create a feature layer to display the shapefile
-        FeatureLayer shapefileFeatureLayer = new FeatureLayer(shapefileFeatureTable);
+        FeatureLayer shapefileFeatureLayer = new FeatureLayer(mShapefileFeatureTable);
```

Comment:
I think we can just move this out of the done loading listener, which would keep the table in scope

Reply:
Yes, I think that is better. Also if you are adding the table to a layer I think you don't need to call `loadAsync()` on the table.